core/controllers/profile_test.py

```diff
@@ -236,11 +237,23 @@ def test_get_profile_picture_valid_username(self):
         response_dict = self.get_json(
             '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
         )
-        # Although the user has a valid username, they have not yet supplied
-        # a profile picture.
-        self.assertIsNone(
+        self.assertIsNotNone(
             response_dict['profile_picture_data_url_for_username'])
 
+    def test_fetch_gravatar(self):
```

Comment:
This test should swap urlfetch.fetch with a function that returns a deterministic string, then compare equality of the resulting profile picture with that deterministic string. That would be more robust than what's currently implemented.

There should also be a test that swaps urlfetch.fetch with something that returns a 404, and in that case the resulting profile picture should be compared with the default identicon data url.